### PR TITLE
Partial support for events in WebXR

### PR DESCRIPTION
**Pull request by Manishearth**
Needs https://github.com/servo/webxr/pull/20

r? @asajeffrey

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23814)

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -5800,6 +5809,7 @@ dependencies = [
 "checksum erased-serde 0.3.9 (registry+https://github.com/rust-lang/crates.io-index)" = "3beee4bc16478a1b26f2e80ad819a52d24745e292f521a63c16eea5f74b7eb60"
 "checksum error-chain 0.12.0 (registry+https://github.com/rust-lang/crates.io-index)" = "07e791d3be96241c77c43846b665ef1384606da2cd2a48730abe606a12906e02"
 "checksum euclid 0.19.8 (registry+https://github.com/rust-lang/crates.io-index)" = "7a4719a544a67ed3fc33784c2bd2c6581663dfe83b719a6ae05c6dabc3b51c73"
+"checksum euclid 0.20.0 (registry+https://github.com/rust-lang/crates.io-index)" = "2c4786521fec183792e755bf32cd0188e4e7628c6d0fcfd51426435b9081a106"
```
Comment (asajeffrey) [inline Cargo.lock]: Hmm, we're dup'ing euclid now? Hopefully the dupe police on onto this!

File: components/script/dom/xrsession.rs [hunk 2]
```diff
@@ -160,17 +232,18 @@ impl XRSession {
...
-                .Context()
-                .Canvas()
-                .upcast::<Node>()
-                .dirty(NodeDamage::OtherNodeDamage);
-        }
+        base_layer
+            .Context()
+            .Canvas()
+            .upcast::<Node>()
+            .dirty(NodeDamage::OtherNodeDamage);
```
Comment (asajeffrey) [inline components/script/dom/xrsession.rs]: This is much nicer now we don't have `XRLayer`!

**Review by asajeffrey — APPROVED** [review]
Looks good!

File: Cargo.lock (hunk 1 above)
Comment (Manishearth) [reply Cargo.lock]: cc @Eijebong

File: components/script/dom/xrsession.rs (hunk 2 above)
Comment (Manishearth) [reply components/script/dom/xrsession.rs]: yay

**Comment by CYBAI** [comment]
```
Diff in /repo/components/script/dom/xrsession.rs at line 395:
         // shuts itself down
         self.ended.set(true);
         global.as_window().Navigator().Xr().end_session(self);
[31m-        self.session
(B[m[31m-            .borrow_mut()
(B[m[31m-            .end_session();
(B[m[32m+        self.session.borrow_mut().end_session();
(B[m         p
     }
 }
Run `./mach fmt` to fix the formatting
```

**Comment by jdm** [comment]
```
{
    "status": "TIMEOUT", 
    "group": "default", 
    "message": "Test timed out", 
    "stack": null, 
    "subtest": "Calling end during an input callback stops processing at the right time", 
    "test": "/webxr/xrSession_input_events_end.https.html", 
    "line": 50304, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "TIMEOUT", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": null, 
    "test": "/webxr/xrSession_input_events_end.https.html", 
    "line": 50305, 
    "action": "test_result", 
    "expected": "ERROR"
}
```

**Comment by jdm** [comment]
Looks like the same failure mode as #23796

**Pull request merged**